app/controllers/CharityNameDetailsController.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import forms.CharityDetailsFormProvider
+import javax.inject.Inject
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.CharityNameDetailsView
+
+import scala.concurrent.Future
+
+class CharityNameDetailsController @Inject()(implicit val appConfig: AppConfig,
+                                             form: CharityDetailsFormProvider,
+                                             view: CharityNameDetailsView,
```

Comment:
In this controller both the view and form are being injected. Not that there's anything wrong, it's just it's inconsistent with how we've done every other controller and I would advise against that.